dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java

```diff
@@ -4913,15 +4915,30 @@ public Node visitWhileLoop(WhileLoopTree tree, Void p) {
             // Condition
             addLabelForNextNode(conditionStart);
             assert tree.getCondition() != null;
+            // Determine whether loop condition is a constant true, according to the compiler logic.
```

Comment:
`true` is a literal not a constant. So this should say `Determine whether loop condition is the boolean literal "true",`.   You've used `constant` instead of `literal` in other places in this pull request.  Please change all uses.

Reply:
More generally, a literal appears in source code, and a constant is an expression that can be evaluated at compile time.  So, `true` is both a literal and a constant.  `4 > 3` is a constant.

Using these terms consistently will help to make the code easier to read.  Thanks!

(I haven't read the code to determine whether it is checking for the literal `true` or the constant `true`.)

Reply:
@smillst This PR correctly uses constant. The compiler handles "!false" and "6 > 4" as being the compile-time constant true.

@d367wang Please add the following test case:

```java
    Issue3249() {
        double local;
        while (6 > 4) {
            local = 1;
            break;
        }
        field = local;
    }
```